### DISCUSSION TITLE
I'm confussed even though I have ML experience

### DISCUSSION THREAD

**Discussion by apavlo89** [asreview/asreview#1422] [Q&A]
Despite all the fancy videos I still don't quite understand what this does and I've got a couple of years of ML experience. Imagine! So what happens after you initially label some papers as relevant or irrelevant? What is the help that this program provides? After labeling some data I go through a review page but there is nowhere that tells me if the ML predicts the paper on display is relevant or not. There are neither predicted ML labels nor confidence intervals of ML label predictions. What is the point?

**Comment by FioByr** [comment]
Hi @apavlo89, thank you for your post!

Well when you start ASReview in oracle mode, you define prior knowledge by labelling at least one paper as relevant, and one as irrelevant. This provides the ML algorithm of your choice with a starting point by which to rank the papers that you will be presented with. After you initially label some papers as relevant and irrelevant, the algorithm will learn from the papers text (titles/abstracts), and will then re-rank the papers that you are presented with in order of relevance. 

The "help that the program provides" is providing you with papers which are predicted to be most relevant to you, and hopefully this will decrease the amount of time needed for you to obtain all the relevant papers, or at least most of them. 

And yes, you are correct, the papers that you are presented with on the review page are not labelled as relevant or irrelevant, they are solely ranked based on their relevance. So that the paper with the highest predicted relevance will be at the beginning of the stack of papers, and the paper with the lowest predicted relevance will be at the end. 

Also, there are no labels nor confidence intervals, as the aim of the software is not to accurately predict relevant papers. Its aim is to decrease the time spent on scanning for relevant papers in the systematic review process. 

I hope this answer helps you understand better what ASReview is attempting to do!

**Comment by rohitgarud** [comment]
Hi @apavlo89,
 I have worked on a few extensions for ASReview, so I have some understanding of how it works under the hood. I will attempt to explain it here:
There are a few components of the ASReview which are based on the active learning principle. Let's see them one by one.
1. **Systematic Literature Review Dataset**:
This dataset contains data related to individual scientific articles/records such as title, abstract, author, year, etc. (ASReview only uses title and abstract by combining title and abstract into a single paragraph text)
2. **Feature Extraction**:
Different feature extraction techniques can be used to convert the text (title + abstract) to a feature vector (like TFIDF, Doc2vec, sentense-BERT, etc) for each record in the dataset. So now we have the feature vectors of each title+abstract in the dataset.
3. **Balancing**:
As there will be more irrelevant records than relevant records, there will be an imbalance between these two classes. So this component uses oversampling to balance the two classes
4. **Classifiers**:
 > What happens after you initially label some papers as relevant or irrelevant?

A classifier such as Naive Bayes, Logistic Regression, SVM, or Random forest will be trained on the feature vectors corresponding to those initially labeled papers. This trained model will be used to classify the remaining unlabelled records (their feature vectors) as relevant or irrelevant. The closer the probability to 1, the record is relevant, the closer the probability to 0, the record is irrelevant.
5. **Query Strategy**:
The query strategy decides which title and abstract to show to the reviewer next so the reviewer can label it relevant or irrelevant. There are a few strategies. The "max query strategy" selects the record having the maximum probability of being relevant. The "uncertainty query strategy" selects the records for which the model is most uncertain (probability around 0.5). "Random query strategy" selects a record at random from the unlabeled records. The "mixed query strategy" combined previously mentioned strategies with a certain probability (default for the mixed criteria is 95% of the time use max query strategy and 5% of the time use random query strategy) which can be adjusted. 

>  After labeling some data I go through a review page but there is nowhere that tells me if the ML predicts the paper on display is relevant or not

After each time the title+abstract is shown to the reviewer and he/she labels it as relevant or irrelevant based on the objective of the SLR. The classifier is again trained using all the labeled records and new probabilities are calculated for the unlabeled records. The process continues. As more and more records are labeled, the classifier gets better at giving a high probability to possible relevant records and a low probability to possible irrelevant records. So only possible relevant records come to the top (in the case of max query strategy) and are shown to the reviewer.
> There are neither predicted ML labels nor confidence intervals of ML label predictions

This is abstracted from the user to avoid bias in their decision of selecting the relevant or irrelevant label for a particular record

>What is the point?

6.**Stopping criteria**:
Say you have 10000 records in the dataset and there are probably less than 100 relevant records. Traditionally, one has to read through all 10000 titles and abstracts to screen them for the next phase of SLR. ASReview will bring the relevant records to the top of the reading list. After you have labeled say 1000 records, you have found 95 relevant records, and say the next 300 records shown by ASReview are consecutively irrelevant. Which might be the time to stop screening with the assumption that there are no relevant records left. At this point, we have essentially reduced the screening time from reading 10000 records to screening only 1300 records in this case. Check [this discussion](https://github.com/asreview/asreview/discussions/557) on stopping criteria.    

You can read more details in the [documentation](https://asreview.readthedocs.io/en/latest/reference.html) and the [official website](https://asreview.nl/blog/active-learning-explained/) 

Hope this helps you. Thank you.

**Reply by apavlo89** [reply]
I finally got it. So a question remains, why don't we get a confidence interval for each paper - I think it would be a useful metric potentially acting as an additional stopping criteria measure.

I would appreciate if anyone latest and greatest best stop criteria practices, or most common one if there is one? Any paper you recommend?

**Reply by rohitgarud** [reply]
I understand what you are proposing here and can potentially help in making a screening stoping decision. But I think we should also consider that it might induce bias as I mentioned, because the reviewer might be influenced by the confidence score (or the probability) given by the algorithm to a specific record being relevant. And the confidence score (probability) is different for various feature/classifier combinations even if we give the same set of prior knowledge records to these combinations.

There are many resources mentioned about the stopping criteria in the discussion I linked above. I have also proposed a [similarity metric-based stopping criteria](https://github.com/asreview/asreview/discussions/1344#discussion-4945510) but it is very crude at the moment without any details of implementation or much evidence of its effectiveness.  Let me know if you have any ideas.

**Reply by rohitgarud** [reply]
ASReview does not even show Authors' names while screening to avoid bias because if the paper is by some popular personality or group in the field, it might get selected solely because of those authors rather than the contents of the title and abstract

**Comment by Rensvandeschoot** [comment]
> I would appreciate if anyone latest and greatest best stop criteria practices, or most common one if there is one? Any paper you recommend?

see the discussion in this thread: https://github.com/asreview/asreview/discussions/557
